packages/flutter/lib/src/material/outline_button.dart

```diff
@@ -0,0 +1,504 @@
+
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder description) {
+    super.debugFillProperties(description);
+    description.add(new ObjectFlagProperty<VoidCallback>('onPressed', onPressed, ifNull: 'disabled'));
+    description.add(new DiagnosticsProperty<ButtonTextTheme>('textTheme', textTheme, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('textColor', textColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('disabledTextColor', disabledTextColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('color', color, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('highlightColor', highlightColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('splashColor', splashColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<double>('highlightElevation', highlightElevation, defaultValue: 2.0));
+    description.add(new DiagnosticsProperty<BorderSide>('borderSide', borderSide, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('disabledBorderColor', disabledBorderColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<Color>('highlightedBorderColor', highlightedBorderColor, defaultValue: null));
+    description.add(new DiagnosticsProperty<EdgeInsetsGeometry>('padding', padding, defaultValue: null));
+    description.add(new DiagnosticsProperty<ShapeBorder>('shape', shape, defaultValue: null));
+  }
+}
+
+class _OutlineButtonState extends State<OutlineButton> with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  Animation<double> _fillAnimation;
+  Animation<double> _elevationAnimation;
+  bool _pressed = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    // The Material widget animates its shape (which includes the outline
+    // border) and elevation over _kElevationDuration. When pressed, the
+    // button makes its fill color opaque white first, and then sets
+    // its highlightElevation. We can't change the elevation while the
+    // button's fill is translucent, because the shadow fills the interior
+    // of the button.
+
+    _controller = new AnimationController(
+      duration: _kPressDuration,
+      vsync: this
+    );
+    _fillAnimation = new CurvedAnimation(
+      parent: _controller,
+      curve: const Interval(0.0, 0.5,
+        curve: Curves.fastOutSlowIn,
+      ),
+    );
+    _elevationAnimation = new CurvedAnimation(
+      parent: _controller,
+      curve: const Interval(0.5, 0.5),
+      reverseCurve: const Interval(1.0, 1.0),
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  ButtonTextTheme _getTextTheme(ButtonThemeData buttonTheme) {
+    return widget.textTheme ?? buttonTheme.textTheme;
+  }
+
+  // TODO(hmuller): this method is the same as FlatButton
```

Comment:
now we have three copies of a bunch of this logic i even more greatly desire to share the code...

Reply:
I will do that in a future PR.